### PR TITLE
Ensure Trusted Code Checkout in GitHub Actions Workflow

### PR DESCRIPTION
**Pull request by UlisesGascon**
## Description

This pull request addresses a potential security issue in our GitHub Actions workflow by ensuring that the code being checked out is from a trusted source. The changes include:

Conditional Ref Checkout:
- Modified the `Git Checkout` step to conditionally use the pull request commit SHA (`github.event.pull_request.head.sha`) only if the event is a pull request.
- For other events, it defaults to using `github.ref`.

## Validation

No local validation was done. This is related to the OSSF Scorecard

## Related Issues

See: https://github.com/nodejs/nodejs.org/pull/6979

### Check List

- [x] I have read the [Contributing Guidelines](https://github.com/nodejs/nodejs.org/blob/main/CONTRIBUTING.md) and made commit messages that follow the guideline.
- [ ] I have run `npm run format` to ensure the code follows the style guide.
- [ ] I have run `npm run test` to check if all tests are passing.
- [ ] I have run `npx turbo build` to check if the website builds without errors.
- [ ] I've covered new added functionality with unit tests if necessary.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -68,7 +68,7 @@ jobs:
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
         with:
           # Provides the Pull Request commit SHA or the GitHub merge group ref
-          ref: ${{ github.event.pull_request.head.sha || github.ref }}
+          ref: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.sha || github.ref }}
```
Comment (avivkeller) [inline .github/workflows/build.yml]:
```suggestion
          ref: ${{ github.event_name == 'pull_request_target' && github.event.pull_request.head.sha || github.ref }}
```

File: .github/workflows/lighthouse.yml [hunk 2]
```diff
@@ -46,7 +46,7 @@ jobs:
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
         with:
           # Provides the Pull Request commit SHA or the GitHub merge group ref
-          ref: ${{ github.event.pull_request.head.sha || github.ref }}
+          ref: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.sha || github.ref }}
```
Comment (avivkeller) [inline .github/workflows/lighthouse.yml]: Ditto

File: .github/workflows/lint-and-tests.yml [hunk 3]
```diff
@@ -73,7 +73,7 @@ jobs:
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
         with:
           # Provides the Pull Request commit SHA or the GitHub merge group ref
-          ref: ${{ github.event.pull_request.head.sha || github.ref }}
+          ref: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.sha || github.ref }}
```
Comment (avivkeller) [inline .github/workflows/lint-and-tests.yml]: Ditto

File: .github/workflows/lint-and-tests.yml [hunk 4]
```diff
@@ -167,7 +167,7 @@ jobs:
...
```
Comment (avivkeller) [inline .github/workflows/lint-and-tests.yml]: Ditto

File: .github/workflows/translations-pr.yml [hunk 5]
```diff
@@ -73,7 +73,7 @@ jobs:
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
         with:
           # Provides the Pull Request commit SHA or the GitHub merge group ref
-          ref: ${{ github.event.pull_request.head.sha || github.ref }}
+          ref: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.sha || github.ref }}
```
Comment (avivkeller) [inline .github/workflows/translations-pr.yml]: Ditto

**Comment by ovflowd** [comment]
@UlisesGascon waiting for you to address the code review changes :)

**Comment by avivkeller** [comment]
For what it's worth, I don't see any security risks with the current setup. If the PR SHA is available, it will be used; otherwise, `github.ref` (which points to the main branch in this case) will be utilized.

The PR SHA can be trusted in this case, as it points to the SHA at the event triggering the workflow execute, meaning that there is no possibility for a race-condition.

**Comment by ovflowd** [comment]
> For what it's worth, I don't see any security risks with the current setup. If the PR SHA is available, it will be used; otherwise, `github.ref` (which points to the main branch in this case) will be utilized.
> 
> The PR SHA can be trusted in this case, as it points to the SHA at the event triggering the workflow execute, meaning that there is no possibility for a race-condition.

I agree here; the `Github` object is reserved; although you can trigger GitHub Actions with custom WebHook payloads, we don't have any configured here. And if that ever happened, this change would be ineffective regardless.

I'm neutral to the change, but I don't believe it is needed.

File: .github/workflows/translations-pr.yml (hunk 5 above)
Comment (UlisesGascon) [reply .github/workflows/translations-pr.yml]: Resolved in https://github.com/nodejs/nodejs.org/pull/7034/commits/0a5dd7421e0913b3804238be8ddd31bf09a49757

File: .github/workflows/lint-and-tests.yml (hunk 4 above)
Comment (UlisesGascon) [reply .github/workflows/lint-and-tests.yml]: Resolved in https://github.com/nodejs/nodejs.org/pull/7034/commits/0a5dd7421e0913b3804238be8ddd31bf09a49757

File: .github/workflows/lint-and-tests.yml (hunk 3 above)
Comment (UlisesGascon) [reply .github/workflows/lint-and-tests.yml]: Resolved in https://github.com/nodejs/nodejs.org/pull/7034/commits/0a5dd7421e0913b3804238be8ddd31bf09a49757

File: .github/workflows/lighthouse.yml (hunk 2 above)
Comment (UlisesGascon) [reply .github/workflows/lighthouse.yml]: Resolved in https://github.com/nodejs/nodejs.org/pull/7034/commits/0a5dd7421e0913b3804238be8ddd31bf09a49757

File: .github/workflows/build.yml (hunk 1 above)
Comment (UlisesGascon) [reply .github/workflows/build.yml]: Resolved in https://github.com/nodejs/nodejs.org/pull/7034/commits/0a5dd7421e0913b3804238be8ddd31bf09a49757

**Comment by UlisesGascon** [comment]
> For what it's worth, I don't see any security risks with the current setup. 

+1 I just want to see if we can remove the warnings from the scorecard 🫤

**Comment by ovflowd** [comment]
> > For what it's worth, I don't see any security risks with the current setup.
> 
> +1 I just want to see if we can remove the warnings from the scorecard 🫤

I am starting to question the real value of the scorecard if it is dumb enough to not recognise false positives.

**Review by ovflowd — APPROVED** [review]
LGTM -- for the sake of our scorecard 🤷

**Pull request merged**